tests/Umbraco.Tests.UnitTests/Umbraco.Core/PropertyEditors/MediaPicker3ValueEditorValidationTests.cs

```diff
@@ -0,0 +1,176 @@
+using System.ComponentModel.DataAnnotations;
+using Moq;
+using NUnit.Framework;
+using Umbraco.Cms.Core.Cache;
+using Umbraco.Cms.Core.IO;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Models.Validation;
+using Umbraco.Cms.Core.PropertyEditors;
+using Umbraco.Cms.Core.Security;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Core.Services.Navigation;
+using Umbraco.Cms.Core.Strings;
+using Umbraco.Cms.Infrastructure.Scoping;
+using Umbraco.Cms.Infrastructure.Serialization;
+
+namespace Umbraco.Cms.Tests.UnitTests.Umbraco.Core.PropertyEditors;
+
+[TestFixture]
+internal class MediaPicker3ValueEditorValidationTests
+{
+    [Test]
+    [TestCase(true, true)]
+    [TestCase(false, true)]
+    [TestCase(false, false)]
+    public void Validates_Start_Node(bool succeed, bool hasAncestorKey)
+    {
+        var (valueEditor, mediaTypeServiceMock, mediaNavigationQueryServiceMock) = CreateValueEditor();
+
+        var parentKey = Guid.NewGuid();
+        IEnumerable<Guid> ancestorKeys = hasAncestorKey ? [Guid.NewGuid()] : [];
+        var mediaKey = Guid.NewGuid();
+
+        if (succeed)
+        {
+            ancestorKeys = [parentKey];
+        }
+
+        mediaNavigationQueryServiceMock.Setup(x => x.TryGetAncestorsKeys(mediaKey, out ancestorKeys)).Returns(true);
+        valueEditor.ConfigurationObject = new MediaPicker3Configuration { StartNodeId = parentKey };
+
+        var value = "[ {\n  \" key\" : \"20266ebe-1f7e-4cf3-a694-7a5fb210223b\",\n  \"mediaKey\" : \"" + mediaKey + "\",\n  \"mediaTypeAlias\" : \"\",\n  \"crops\" : [ ],\n  \"focalPoint\" : null\n} ]";
+
+        var result = valueEditor.Validate(value, false, null, PropertyValidationContext.Empty());
+
+        ValidateResult(succeed, result, "invalidStartNode");
+    }
+
+    [Test]
+    [TestCase(true, true)]
+    [TestCase(false, true)]
+    [TestCase(false, false)]
+    public void Validates_Allowed_Type(bool succeed, bool findsMediaType)
```

Comment:
I find the parameters here a little confusing, as it looks like `succeed` is doing two things - affecting what you mock, and in the assert for the result you check.  Maybe it's valid from a test perspective, but perhaps clearer if you used three parameters, e.g.:

```
public void Validates_Allowed_Type(bool hasAllowedType, bool findsMediaType, bool expectSuccess)
```